src/finch/codegen/safe_buffer.py

```diff
@@ -0,0 +1,160 @@
+import ctypes
+from typing import NamedTuple
+
+import numpy as np
+
+from finch.codegen.numpy_buffer import (
+    BufferFields,
+    CNumpyBuffer,
+    NumpyBufferFType,
+    NumpyBuffer,
+)
+
+import finch.finch_assembly as asm
+from finch.finch_assembly import Buffer
+from finch.util import qual_str
+from finch.codegen.c import CBufferFType, CCompiler, CContext, CStackFType, c_type
+from finch.codegen.numba_backend import NumbaBufferFType
+
+
+def SafeBuffer(
```

Comment:
why a factory instead of a wrapper class like in https://github.com/finch-tensor/finch-tensor-lite/pull/144? I think it would be more straightforward to define
```
class SafeBuffer(AbstractBuffer...):
    def __init__(self, child):
        self.child_buffer = child
    def load(self, idx):
        assert idx < self.child_buffer
        return self.child_buffer.load
```